packages/wp-now/src/config/providers/WpEnvConfigProvider.ts

```diff
@@ -0,0 +1,105 @@
+import * as fs from 'fs';
+import path from 'path';
+import { CliOptions } from './CliConfigProvider';
+import { WPEnvOptions } from './ConfigProvider';
+import { ConfigProviderInterface, WPNowOptions } from './ConfigProvider';
+
+const WP_ENV_FILE = '.wp-env.json';
+const WP_ENV_OVERRIDE_FILE = '.wp-env.override.json';
+
+const WP_ENV_TO_WP_NOW_MAPPER = {
+	phpVersion: 'phpVersion',
+	port: 'port',
+};
+
+class WpEnvConfigProvider implements ConfigProviderInterface {
+	priority: 1;
+
+	#options: CliOptions | null;
+	constructor(options: CliOptions) {
+		this.#options = options;
+	}
+
+	#mergeConfigs(
+		wpEnvConfig: WPEnvOptions,
+		wpOverrideEnvConfig: WPEnvOptions
+	): WPNowOptions {
+		if (!wpEnvConfig && !wpOverrideEnvConfig) {
+			return null;
+		}
+
+		if (!wpEnvConfig && wpOverrideEnvConfig) {
+			return wpOverrideEnvConfig;
+		}
+
+		const config: WPNowOptions = {};
+		for (const [key, value] of Object.entries(wpEnvConfig || {})) {
+			if (wpOverrideEnvConfig && wpOverrideEnvConfig[key]) {
+				config[key] = wpOverrideEnvConfig[key];
+				continue;
+			}
+			config[key] = value;
+		}
+		return config;
+	}
+
+	#getConfigPath(configName: string): string {
+		return path.join(
+			(this.#options.path as string) || process.cwd(),
+			configName
+		);
+	}
+
+	configExists(): boolean {
+		const wpEnvConfigPath = this.#getConfigPath(WP_ENV_FILE);
+		const wpEnvOverrideConfigPath =
+			this.#getConfigPath(WP_ENV_OVERRIDE_FILE);
+		return (
+			fs.existsSync(wpEnvConfigPath) ||
+			fs.existsSync(wpEnvOverrideConfigPath)
+		);
+	}
+
+	#toWpNow(wpenvConfig: WPEnvOptions): WPNowOptions {
+		if (!wpenvConfig) {
+			return null;
+		}
+		const config: WPNowOptions = {};
+		for (const [key, value] of Object.entries(wpenvConfig)) {
+			if (WP_ENV_TO_WP_NOW_MAPPER[key]) {
+				config[WP_ENV_TO_WP_NOW_MAPPER[key]] = value;
+			}
+		}
+		return config;
+	}
+	public async getConfig(): Promise<WPNowOptions> {
+		if (!this.configExists()) {
+			return null;
+		}
+		let wpEnvConfig = null;
+		try {
+			wpEnvConfig = this.#toWpNow(
+				JSON.parse(
+					fs.readFileSync(this.#getConfigPath(WP_ENV_FILE), 'utf8')
+				)
+			);
+			// eslint-disable-next-line no-empty
+		} catch (error) {}
+
+		let wpOverrideEnvConfig = null;
+		try {
+			wpOverrideEnvConfig = this.#toWpNow(
+				JSON.parse(
+					fs.readFileSync(
+						this.#getConfigPath(WP_ENV_OVERRIDE_FILE),
+						'utf8'
+					)
+				)
+			);
+			// eslint-disable-next-line no-empty
+		} catch {}
+		return this.#mergeConfigs(wpEnvConfig, wpOverrideEnvConfig);
```

Comment:
So we convert wp env config to wp now format and then merge them. Is there any utility to merge wp-env configs? This way we'd deal with the final config and wouldn't have to maintain any merging logic.

Reply:
Do you mean a util that already exists in some other codebase and that can be imported?
I read `wp-env`'s code, and it seems they are doing something similar. 
Do you have any suggestions here?
To me, the current logic doesn't seem to have a need of any special maintenance other than add new fields as we go.
This is because the two configs are the same, adding just a new property to `WP_ENV_TO_WP_NOW_MAPPER` is the only change we should probably make in the future.

Reply:
How about a function like `fromWpEnvConfig(path)` that encapsulates the try/catch? It should make this function much smaller